sdk/include/FreeRTOS-Compat/semphr.h

```diff
@@ -1,18 +1,141 @@
 #pragma once
+/**
+ * FreeRTOS semaphore compatibility layer.  This maps FreeRTOS semaphore types
+ * to their CHERIoT RTOS equivalents.
+ *
+ * There is some overhead from dynamic dispatch that can be avoided if only one
+ * of the FreeRTOS semaphore types needs to be supported in a particular
+ * component.  You can enable individual semaphore types by defining the
+ * following macros:
+ *
+ * - `CHERIOT_EXPOSE_FREERTOS_SEMAPHORE`: Enable counting and binary semaphores.
+ * - `CHERIOT_EXPOSE_FREERTOS_MUTEX`: Enable non-recursive, priority-inheriting
+ * mutexes.
+ * - `CHERIOT_EXPOSE_FREERTOS_RECURSIVE_MUTEX`: Enable recursive mutexes.
+ */
 #include "FreeRTOS.h"
 #include <locks.h>
 #include <stdatomic.h>
 
+#if !defined(CHERIOT_EXPOSE_FREERTOS_SEMAPHORE) &&                             \
+  !defined(CHERIOT_EXPOSE_FREERTOS_MUTEX) &&                                   \
+  !defined(CHERIOT_EXPOSE_FREERTOS_RECURSIVE_MUTEX)
+#	error At least one out of CHERIOT_EXPOSE_FREERTOS_SEMAPHORE, \
+  CHERIOT_EXPOSE_FREERTOS_MUTEX, or CHERIOT_EXPOSE_FREERTOS_RECURSIVE_MUTEX \
+  must be defined.
+#endif
+
+#ifdef CHERIOT_EXPOSE_FREERTOS_SEMAPHORE
+#	define CHERIOT_FREERTOS_SEMAPHORE 1
```

Comment:
While I don't object to these `EXPOSE`-less `define`s, are they doing anything that `defined(CHERIOT_EXPOSE_...)` can't?

Reply:
FWIW, `defined()` returns `1` or `0`, so I think you can `defined() + defined() + ...`

Reply:
Ah, I forgot that C thinks Booleans are integers. That should simplify it slightly.

Reply:
Fixed, hopefully ready to merge now.